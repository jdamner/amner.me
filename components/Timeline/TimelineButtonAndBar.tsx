import React from "react";

export default function TimelineButtonAndBar({
  last = false,
}: {
  last?: boolean;
}) {
  const buttonClasses = [
    "absolute",
    "w-4",
    "h-4",
    "mt-2.5",
    "-top-2.5",
    "bg-slate-900",
    "dark:bg-slate-400",
    "rounded-full",
  ];

  const barClasses = ["mt-1.5", "w-full", "h-1"];

  if (last) {
    barClasses.push(
      "bg-linear-to-r from-slate-300 to-transparent dark:from-slate-600",
    );
  } else {
    barClasses.push("bg-slate-300 dark:bg-slate-600");
  }
  return (
    <>
      <div className={barClasses.join(" ")} />
      <div className={buttonClasses.join(" ")} />
    </>
  );
}
