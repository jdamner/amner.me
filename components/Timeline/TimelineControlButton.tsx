import React from "react";

export default function TimelineControlButton(
  props: React.DetailedHTMLProps<
    React.ButtonHTMLAttributes<HTMLButtonElement>,
    HTMLButtonElement
  > & { direction?: "left" | "right" },
) {
  const { direction = "left", ...rest } = props;

  const arrow =
    direction === "left" ? (
      <svg
        className="w-6 h-6"
        fill="none"
        stroke="currentColor"
        viewBox="0 0 24 24"
      >
        <path
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          d="M15 19l-7-7 7-7"
        />
      </svg>
    ) : (
      <svg
        className="w-6 h-6"
        fill="none"
        stroke="currentColor"
        viewBox="0 0 24 24"
      >
        <path
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          d="M9 5l7 7-7 7"
        />
      </svg>
    );

  return (
    <button
      className="p-2 w-10 h-10 mx-3
                bg-orange-600 text-orange-100
                hover:bg-white hover:text-black
                hover:cursor-pointer
                disabled:opacity-50 disabled:cursor-default"
      {...rest}
    >
      {arrow}
    </button>
  );
}
