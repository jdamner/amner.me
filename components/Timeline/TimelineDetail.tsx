import React from "react";

export default function TimelineDetail({ children }: React.PropsWithChildren) {
  return <div className="prose px-3 md:px-0 prose-invert">{children}</div>;
}
