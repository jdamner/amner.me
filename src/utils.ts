/**
 * Helper functions
 */
import fs from 'fs'
import matter from 'gray-matter'
import { join } from 'path'
import { slug } from 'github-slugger'

import type { ReactNode } from 'react'
import type { MdFile } from './types'

/**
 * Makes an object JSON parseable
 * 
 * @param {any} obj The object to be parsed
 * @returns {any}
 */
export function makeJsonParseable<Type>(obj: Type): Type {
	return JSON.parse(JSON.stringify(obj)) as Type
}

/**
 * Reads a markdown file
 */
export async function readMdFile(filename: string, directory: string = 'content/'): Promise<MdFile> {
	const path = `${directory}/${filename}`
	const fullPath = `${process.cwd()}/${path}`
	const fileContents = fs.readFileSync(fullPath, 'utf8')
	return {
		slug: slug(filename.replace(/\.md$/, '')),
		...matter(fileContents),
	}
}

/**
 * Reads all slugs from a directory
 * 
 * @param {string} directory The directory to read from
 * 
 * @returns {string[]}
 */
export function getFilesInDir(directory: string): string[] {
	const fullPath = join(process.cwd(), directory)
	return fs.readdirSync(fullPath)
}

/**
 * Converts all headings in a markdown file to have IDs 
 * Converts 
 * ## Heading 1 
 * to become
 * ## Heading 1{#heading-1}
 * 
 * @param {string} markdown The markdown to convert
 * 
 * @returns {string}
 */
export function convertHeadingsToAnchorLinks(markdown: string): string {
	return markdown.replace(/(#{1,6})\s(.+)\n/gm, (match, p1, p2) => {
		return `${p1} ${p2}{#${slug(p2)}}\n`
	})
}


/**
 * Converts a react node to a slug
 * 
 * Usually the value is a string, but sometimes it contains
 * a `<code>` element, which is a react node. This function
 * converts the value to a string.
 */
export function makeSlug(value: ReactNode | ReactNode[]): string {
	if (! Array.isArray(value)) return slug(value?.toString() ?? '');
	return slug(
		value.map((node) => {
			if (typeof node === 'string' ||
				typeof node === 'number' ||
				typeof node === 'boolean'
			) return node;

			if (typeof node === 'object' && node !== null && 'props' in node) {
				return makeSlug(node.props.children)
			}

			return ''
		}).join('')
	)
}