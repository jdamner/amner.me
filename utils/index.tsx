/* External Deps */
import { readFileSync, readdirSync } from "fs";
import { resolve } from "path";
import matter from "gray-matter";
import { slug } from "github-slugger";

/* Components */
import { Markdown } from "@/components/Markdown";

/* Constants */
const postsDirectory: string = resolve("content/blog/");
const servicesDirectory: string = resolve("content/services");
const cvDirectory: string = resolve("content/cv");
const projectsDirectory: string = resolve(cvDirectory, "projects");
const referencesDirectory: string = resolve(cvDirectory, "references");
const employmentDirectory: string = resolve(cvDirectory, "employment");
const educationDirectory: string = resolve(cvDirectory, "education");

type MatterData = ReturnType<typeof matter>["data"];
type AttributeValidator<T extends MatterData> = (data: MatterData) => data is T;

const defaultAttributeValidator = <T extends MatterData>(
  data: MatterData,
): data is T => {
  return typeof data === "object" && data !== null;
};

/**
 * Gets all files in a directory
 */
const getAllFrom = <T extends MatterData>(
  directory: string,
  attributeValidator: AttributeValidator<T> = defaultAttributeValidator,
) =>
  readdirSync(directory).map((filename) =>
    readMdFile<T>(filename, directory, attributeValidator),
  );

/**
 * Reads a markdown file
 */
const readMdFile = <T extends MatterData>(
  filename: string,
  directory: string = "content/",
  attributeValidator: AttributeValidator<T> = defaultAttributeValidator,
) =>
  parseMarkdown<T>(
    readFileSync(`${directory}/${filename}`, "utf8"),
    filename,
    attributeValidator,
  );

/**
 * Parses markdown file contents
 */
export const parseMarkdown = <T extends MatterData>(
  fileContents: string,
  filename: string,
  attributeValidator: AttributeValidator<T> = defaultAttributeValidator,
) => {
  const { content, data } = matter(fileContents);

  if (!attributeValidator(data)) {
    console.log(data);
    throw new Error(`Invalid attributes in ${filename}` + JSON.stringify(data));
  }

  return {
    data,
    slug: slug(filename.replace(/\.mdx$/, "")),
    content: <Markdown>{content}</Markdown>,
  };
};

/**
 * Gets all Posts
 */
const getAllPosts = () =>
  getAllFrom<WithTitle & WithThumbnail & WithDate>(
    postsDirectory,
    (data): data is WithTitle & WithThumbnail & WithDate =>
      hasTitle(data) && hasThumbnail(data) && hasDate(data),
  );

export const useAllServices = () =>
  getAllFrom<WithTitle>(servicesDirectory, hasTitle);

export const getAllProjects = () =>
  getAllFrom<WithTitle & WithLink>(
    projectsDirectory,
    (data): data is WithTitle & WithLink => hasTitle(data) && hasLink(data),
  );

export const useAllReferences = () =>
  getAllFrom<WithTitle & WithWebsite & WithPhone & WithEmail>(
    referencesDirectory,
    (data): data is WithTitle & WithWebsite & WithPhone & WithEmail =>
      hasTitle(data) && hasWebsite(data) && hasPhone(data) && hasEmail(data),
  );

export const useAllEmployment = () =>
  getAllFrom<
    WithTitle & WithSubtitle & WithDateString & WithRole & WithWebsite
  >(
    employmentDirectory,
    (
      data,
    ): data is WithTitle &
      WithSubtitle &
      WithDateString &
      WithRole &
      WithWebsite =>
      hasTitle(data) &&
      hasSubtitle(data) &&
      hasDateString(data) &&
      hasRole(data) &&
      hasWebsite(data),
  );

export const useAllEducation = () =>
  getAllFrom<WithTitle & WithDateString>(
    educationDirectory,
    (data): data is WithTitle & WithDateString =>
      hasTitle(data) && hasDateString(data),
  );

export const getPost = (slug: string) =>
  getAllPosts().find((post) => post.slug === slug);

export const getAllPostLinks = () =>
  getAllPosts()
    .map((post) => ({
      title: post.data.title,
      thumbnail: post.data.thumbnail,
      slug: post.slug,
      date: post.data.date,
    }))
    .sort((a, b) => (a.date > b.date ? -1 : 1));

type WithTitle = { title: string };
type WithSubtitle = { subtitle: string };
type WithLink = { link: string };
type WithThumbnail = { thumbnail: string };
type WithDate = { date: Date };
type WithDateString = { date: string };
type WithPhone = { phone: string };
type WithWebsite = { website: string };
type WithEmail = { email: string };
type WithRole = { role: string };

const hasTitle = (data: MatterData): data is WithTitle =>
  defaultAttributeValidator(data) &&
  "title" in data &&
  typeof data.title === "string";

const hasSubtitle = (data: MatterData): data is WithSubtitle =>
  "subtitle" in data && typeof data.subtitle === "string";

const hasLink = (data: MatterData): data is WithLink =>
  "link" in data && typeof data.link === "string";

const hasThumbnail = (data: MatterData): data is WithThumbnail =>
  "thumbnail" in data && typeof data.thumbnail === "string";

const hasDate = (data: MatterData): data is WithDate =>
  "date" in data && data.date instanceof Date;

const hasDateString = (data: MatterData): data is WithDateString =>
  "date" in data && typeof data.date === "string";

const hasPhone = (data: MatterData): data is WithPhone =>
  "phone" in data && typeof data.phone === "string";

const hasWebsite = (data: MatterData): data is WithWebsite =>
  "website" in data && typeof data.website === "string";

const hasEmail = (data: MatterData): data is WithEmail =>
  "email" in data && typeof data.email === "string";

export const hasRole = (data: MatterData): data is WithRole =>
  "role" in data && typeof data.role === "string";
