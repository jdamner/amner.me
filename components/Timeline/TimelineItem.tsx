import React from "react";

import InlineLink from "../Atoms/Links/Inline";
import TimelineButtonAndBar from "./TimelineButtonAndBar";

export default function TimelineItem(
  props: React.DetailedHTMLProps<
    React.LiHTMLAttributes<HTMLLIElement>,
    HTMLLIElement
  > &
    Partial<{
      role: string;
      company: string;
      companyUrl: string;
      date: string;
      last: boolean;
    }>,
) {
  const { role, company, companyUrl, date, last = false, ...rest } = props;
  return (
    <li
      className="group snap-always snap-start  shrink-0 w-9/12 last:w-full"
      {...rest}
    >
      <TimelineButtonAndBar last={last} />
      <div className="mt-3 text-slate-900 dark:text-slate-200">
        <h3 className="w-full text-left text-2xl font-black mb-2">{role}</h3>
        <h4 className="text-lg">
          <InlineLink href={companyUrl} target="_blank">
            {company}
          </InlineLink>
        </h4>
        <span className="text-sm">{date}</span>
      </div>
    </li>
  );
}
