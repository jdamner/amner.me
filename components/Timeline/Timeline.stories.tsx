import type { Meta, StoryObj } from "@storybook/react";

import { faker } from "@faker-js/faker";
import Timeline from "./Timeline";
import { generateMdFile } from "../../utils/storybook";

type Story = StoryObj<typeof Timeline>;

export default {
  title: "Components/Timeline",
  component: Timeline,
} as Meta;

faker.seed(123);

export const Default: Story = {
  args: {
    events: Array.from({ length: 10 }, () =>
      generateMdFile({
        role: faker.person.jobTitle(),
        title: faker.company.name(),
        url: faker.internet.url(),
        from: "March 2020", // faker dates aren't consistent even with a seed
        to: "Present",
      }),
    ),
  },
};
