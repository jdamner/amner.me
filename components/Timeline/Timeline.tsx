"use client";

import React, { useState } from "react";
import TimelineItem from "./TimelineItem";
import TimelineDetail from "./TimelineDetail";
import TimelineControlButton from "./TimelineControlButton";

import type { MdFile } from "../../types";

export default function Timeline(
  props: React.DetailedHTMLProps<
    React.HTMLAttributes<HTMLDivElement>,
    HTMLDivElement
  > & { events: MdFile[] },
) {
  const { events, ...rest } = props;

  const [currentIndex, setCurrentIndex] = useState<number>(0);
  const sliderRef = React.useRef<HTMLOListElement>(null);

  const handleScroll = (event: React.UIEvent<HTMLOListElement, UIEvent>) => {
    const { clientWidth } = event.currentTarget;
    let index = 0;

    for (const child of event.currentTarget.children) {
      const childLeft = child.getBoundingClientRect().left;
      const childWidth = child.getBoundingClientRect().width;

      if (
        index !== currentIndex && // If the index is not the current index
        index <= events?.length && // If the index is not greater than the number of events
        childLeft < clientWidth / 2 && // If the left side of the child is less than half the width of the screen
        childLeft + childWidth > clientWidth / 2 // If the right side of the child is greater than half the width of the screen
      ) {
        setCurrentIndex(index);
      }
      index++;
    }
  };

  const forceScrollTo = (index: number) => {
    const element = sliderRef.current;
    element?.children[index]?.scrollIntoView({
      behavior: "smooth",
      block: "nearest",
      inline: "start",
    });
  };

  return (
    <div className="flex flex-col" {...rest}>
      {events?.length > 0 && (
        <>
          <ol
            ref={sliderRef}
            className="relative w-full flex snap-x snap-mandatory overflow-x-auto no-scrollbar"
            onScroll={handleScroll}
          >
            {events.map((event, index) => (
              <TimelineItem
                event={event}
                key={event.slug + index.toString()}
                last={index === events.length - 1}
              />
            ))}
          </ol>
          <div className="ml-auto my-3">
            <TimelineControlButton
              name={"Previous"}
              aria-label="Previous event"
              direction="left"
              onClick={() => forceScrollTo(currentIndex - 1)}
              disabled={currentIndex === 0}
            />
            <TimelineControlButton
              name={"Next"}
              aria-label="Next event"
              direction="right"
              onClick={() => forceScrollTo(currentIndex + 1)}
              disabled={currentIndex === events.length - 1}
            />
          </div>
          <div className="flex justify-between flex-col md:flex-row">
            {events[currentIndex] && (
              <TimelineDetail event={events[currentIndex]} />
            )}
          </div>
        </>
      )}
    </div>
  );
}
