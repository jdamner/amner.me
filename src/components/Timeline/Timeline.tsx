import React, { useState } from "react";
import TimelineItem from "./TimelineItem";
import TimelineDetail from "./TimelineDetail";
import TimelineControlButton from "./TimelineControlButton";

import type { MdFile } from "../../types/MdFile.type";

export default function Timeline({ events }: { events: MdFile[] }): React.JSX.Element {

    const [currentIndex, setCurrentIndex] = useState<number>(0)

    const sliderRef = React.useRef<HTMLOListElement>(null);

    const handleScroll = (event: React.UIEvent<HTMLOListElement, UIEvent>) => {
        const clientWidth = event.currentTarget.clientWidth
        let index = 0

        for (let child of event.currentTarget.children) {
            let childLeft = child.getBoundingClientRect().left
            let childWidth = child.getBoundingClientRect().width

            if (
                index !== currentIndex && // If the index is not the current index
                index <= events?.length && // If the index is not greater than the number of events
                childLeft < clientWidth / 2 && // If the left side of the child is less than half the width of the screen
                childLeft + childWidth > clientWidth / 2 // If the right side of the child is greater than half the width of the screen
            ) {
                setCurrentIndex(index)
            }
            index++;
        }
    }

    const forceScrollTo = (index: number) => {
        const element = sliderRef.current
        element?.children[index]?.scrollIntoView({
            behavior: 'smooth',
            block: 'nearest',
            inline: 'start',
        })
    }

    return (
        <div className='flex flex-col'>
            {events?.length > 0 &&
                <>
                    <ol ref={sliderRef} className="relative w-full flex snap-x snap-mandatory overflow-x-auto no-scrollbar" onScroll={handleScroll}>
                        {events.map((event, index) => (
                            <TimelineItem event={event} key={event.slug} last={index === events.length - 1} />
                        ))}
                    </ol>
                    <div className="ml-auto my-3">
                            <TimelineControlButton direction="left" onClick={() => forceScrollTo(currentIndex - 1)} disabled={currentIndex === 0} />
                            <TimelineControlButton direction="right" onClick={() => forceScrollTo(currentIndex + 1)} disabled={currentIndex === events.length - 1} />
                        </div>
                    <div className="flex justify-between flex-col md:flex-row">
                        {events[currentIndex] && <TimelineDetail event={events[currentIndex]} />}

                        

                    </div>
                </>
            }
        </div>
    )
}

