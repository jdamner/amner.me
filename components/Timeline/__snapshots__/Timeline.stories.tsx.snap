// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Timeline Default smoke-test 1`] = `
<div class="bg-slate-100 dark:bg-slate-900 text-slate-900 dark:text-slate-300">
  <div class="flex flex-col">
    <ol class="relative w-full flex snap-x snap-mandatory overflow-x-auto no-scrollbar">
      <li class="group snap-always snap-start  shrink-0 w-9/12 last:w-full"
          title="mesh synergistic users"
      >
        <div class="mt-1.5 w-full h-1 bg-slate-300 dark:bg-slate-600">
        </div>
        <div class="absolute w-4 h-4 mt-2.5 -top-2.5 bg-slate-900 dark:bg-slate-400 rounded-full">
        </div>
        <div class="mt-3 text-slate-900 dark:text-slate-200">
          <h3 class="w-full text-left text-2xl font-black mb-2">
            International Assurance Specialist
          </h3>
          <h4 class="text-lg">
            <a class="underline hover:text-slate-800 hover:no-underline"
               target="_blank"
               href="https://impressive-script.org/"
            >
              Farrell - Pollich
            </a>
          </h4>
          <span class="text-sm">
            March 2020 - Present
          </span>
        </div>
      </li>
      <li class="group snap-always snap-start  shrink-0 w-9/12 last:w-full"
          title="grow frictionless channels"
      >
        <div class="mt-1.5 w-full h-1 bg-slate-300 dark:bg-slate-600">
        </div>
        <div class="absolute w-4 h-4 mt-2.5 -top-2.5 bg-slate-900 dark:bg-slate-400 rounded-full">
        </div>
        <div class="mt-3 text-slate-900 dark:text-slate-200">
          <h3 class="w-full text-left text-2xl font-black mb-2">
            Future Security Agent
          </h3>
          <h4 class="text-lg">
            <a class="underline hover:text-slate-800 hover:no-underline"
               target="_blank"
               href="https://awesome-rhythm.com"
            >
              McDermott - Schultz
            </a>
          </h4>
          <span class="text-sm">
            March 2020 - Present
          </span>
        </div>
      </li>
      <li class="group snap-always snap-start  shrink-0 w-9/12 last:w-full"
          title="matrix visionary e-markets"
      >
        <div class="mt-1.5 w-full h-1 bg-slate-300 dark:bg-slate-600">
        </div>
        <div class="absolute w-4 h-4 mt-2.5 -top-2.5 bg-slate-900 dark:bg-slate-400 rounded-full">
        </div>
        <div class="mt-3 text-slate-900 dark:text-slate-200">
          <h3 class="w-full text-left text-2xl font-black mb-2">
            Human Division Supervisor
          </h3>
          <h4 class="text-lg">
            <a class="underline hover:text-slate-800 hover:no-underline"
               target="_blank"
               href="https://lost-parable.com/"
            >
              Kulas, McGlynn and Abbott
            </a>
          </h4>
          <span class="text-sm">
            March 2020 - Present
          </span>
        </div>
      </li>
      <li class="group snap-always snap-start  shrink-0 w-9/12 last:w-full"
          title="unleash extensible communities"
      >
        <div class="mt-1.5 w-full h-1 bg-slate-300 dark:bg-slate-600">
        </div>
        <div class="absolute w-4 h-4 mt-2.5 -top-2.5 bg-slate-900 dark:bg-slate-400 rounded-full">
        </div>
        <div class="mt-3 text-slate-900 dark:text-slate-200">
          <h3 class="w-full text-left text-2xl font-black mb-2">
            Lead Usability Technician
          </h3>
          <h4 class="text-lg">
            <a class="underline hover:text-slate-800 hover:no-underline"
               target="_blank"
               href="https://breakable-boxspring.name/"
            >
              Batz - Hirthe
            </a>
          </h4>
          <span class="text-sm">
            March 2020 - Present
          </span>
        </div>
      </li>
      <li class="group snap-always snap-start  shrink-0 w-9/12 last:w-full"
          title="recontextualize virtual mindshare"
      >
        <div class="mt-1.5 w-full h-1 bg-slate-300 dark:bg-slate-600">
        </div>
        <div class="absolute w-4 h-4 mt-2.5 -top-2.5 bg-slate-900 dark:bg-slate-400 rounded-full">
        </div>
        <div class="mt-3 text-slate-900 dark:text-slate-200">
          <h3 class="w-full text-left text-2xl font-black mb-2">
            International Applications Engineer
          </h3>
          <h4 class="text-lg">
            <a class="underline hover:text-slate-800 hover:no-underline"
               target="_blank"
               href="https://diligent-chrome.com/"
            >
              Kassulke, Hauck and MacGyver
            </a>
          </h4>
          <span class="text-sm">
            March 2020 - Present
          </span>
        </div>
      </li>
      <li class="group snap-always snap-start  shrink-0 w-9/12 last:w-full"
          title="disintermediate cross-media e-markets"
      >
        <div class="mt-1.5 w-full h-1 bg-slate-300 dark:bg-slate-600">
        </div>
        <div class="absolute w-4 h-4 mt-2.5 -top-2.5 bg-slate-900 dark:bg-slate-400 rounded-full">
        </div>
        <div class="mt-3 text-slate-900 dark:text-slate-200">
          <h3 class="w-full text-left text-2xl font-black mb-2">
            National Communications Coordinator
          </h3>
          <h4 class="text-lg">
            <a class="underline hover:text-slate-800 hover:no-underline"
               target="_blank"
               href="https://adolescent-in-joke.org"
            >
              Hagenes - Stanton
            </a>
          </h4>
          <span class="text-sm">
            March 2020 - Present
          </span>
        </div>
      </li>
      <li class="group snap-always snap-start  shrink-0 w-9/12 last:w-full"
          title="exploit enterprise niches"
      >
        <div class="mt-1.5 w-full h-1 bg-slate-300 dark:bg-slate-600">
        </div>
        <div class="absolute w-4 h-4 mt-2.5 -top-2.5 bg-slate-900 dark:bg-slate-400 rounded-full">
        </div>
        <div class="mt-3 text-slate-900 dark:text-slate-200">
          <h3 class="w-full text-left text-2xl font-black mb-2">
            Senior Tactics Architect
          </h3>
          <h4 class="text-lg">
            <a class="underline hover:text-slate-800 hover:no-underline"
               target="_blank"
               href="https://unselfish-honor.name/"
            >
              Kilback - Ankunding
            </a>
          </h4>
          <span class="text-sm">
            March 2020 - Present
          </span>
        </div>
      </li>
      <li class="group snap-always snap-start  shrink-0 w-9/12 last:w-full"
          title="streamline next-generation interfaces"
      >
        <div class="mt-1.5 w-full h-1 bg-slate-300 dark:bg-slate-600">
        </div>
        <div class="absolute w-4 h-4 mt-2.5 -top-2.5 bg-slate-900 dark:bg-slate-400 rounded-full">
        </div>
        <div class="mt-3 text-slate-900 dark:text-slate-200">
          <h3 class="w-full text-left text-2xl font-black mb-2">
            Senior Factors Associate
          </h3>
          <h4 class="text-lg">
            <a class="underline hover:text-slate-800 hover:no-underline"
               target="_blank"
               href="https://fancy-parka.biz/"
            >
              Parisian - Klein
            </a>
          </h4>
          <span class="text-sm">
            March 2020 - Present
          </span>
        </div>
      </li>
      <li class="group snap-always snap-start  shrink-0 w-9/12 last:w-full"
          title="repurpose wireless interfaces"
      >
        <div class="mt-1.5 w-full h-1 bg-slate-300 dark:bg-slate-600">
        </div>
        <div class="absolute w-4 h-4 mt-2.5 -top-2.5 bg-slate-900 dark:bg-slate-400 rounded-full">
        </div>
        <div class="mt-3 text-slate-900 dark:text-slate-200">
          <h3 class="w-full text-left text-2xl font-black mb-2">
            Corporate Implementation Supervisor
          </h3>
          <h4 class="text-lg">
            <a class="underline hover:text-slate-800 hover:no-underline"
               target="_blank"
               href="https://automatic-hearing.info/"
            >
              Tromp - Jerde
            </a>
          </h4>
          <span class="text-sm">
            March 2020 - Present
          </span>
        </div>
      </li>
      <li class="group snap-always snap-start  shrink-0 w-9/12 last:w-full"
          title="iterate dot-com web services"
      >
        <div class="mt-1.5 w-full h-1 bg-gradient-to-r from-slate-300 to-transparent dark:from-slate-600">
        </div>
        <div class="absolute w-4 h-4 mt-2.5 -top-2.5 bg-slate-900 dark:bg-slate-400 rounded-full">
        </div>
        <div class="mt-3 text-slate-900 dark:text-slate-200">
          <h3 class="w-full text-left text-2xl font-black mb-2">
            Dynamic Usability Specialist
          </h3>
          <h4 class="text-lg">
            <a class="underline hover:text-slate-800 hover:no-underline"
               target="_blank"
               href="https://earnest-wafer.org/"
            >
              Shields, Sipes and Pollich
            </a>
          </h4>
          <span class="text-sm">
            March 2020 - Present
          </span>
        </div>
      </li>
    </ol>
    <div class="ml-auto my-3">
      <button class="p-2 w-10 h-10 mx-3 bg-slate-400 hover:bg-slate-600 dark:hover:bg-slate-200 text-white dark:text-slate-900 hover:cursor-pointer disabled:opacity-50 disabled:cursor-default disabled:hover:bg-slate-400"
              name="Previous"
              aria-label="Previous event"
              disabled
      >
        <svg class="w-6 h-6"
             fill="none"
             stroke="currentColor"
             viewbox="0 0 24 24"
        >
          <path stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                d="M15 19l-7-7 7-7"
          >
          </path>
        </svg>
      </button>
      <button class="p-2 w-10 h-10 mx-3 bg-slate-400 hover:bg-slate-600 dark:hover:bg-slate-200 text-white dark:text-slate-900 hover:cursor-pointer disabled:opacity-50 disabled:cursor-default disabled:hover:bg-slate-400"
              name="Next"
              aria-label="Next event"
      >
        <svg class="w-6 h-6"
             fill="none"
             stroke="currentColor"
             viewbox="0 0 24 24"
        >
          <path stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                d="M9 5l7 7-7 7"
          >
          </path>
        </svg>
      </button>
    </div>
    <div class="flex justify-between flex-col md:flex-row">
      <div class="prose px-3 md:px-0 prose-slate dark:prose-invert">
        <h1 id="cunctatio-stillicidium-nisi"
            node="[object Object]"
        >
          cunctatio stillicidium nisi
        </h1>
        <h2 id="denique-apud"
            node="[object Object]"
        >
          denique apud
        </h2>
        <p>
          Cui conicio quis suspendo commodo decens. Adulatio sub cubicularis paens. Ratione auctor decerno atque agnosco exercitationem summa exercitationem.
        </p>
        <h3 id="cursus-reiciendis"
            node="[object Object]"
        >
          cursus reiciendis
        </h3>
        <p>
          Ulciscor deprimo surculus taedium perferendis. Suppono cur commodi sollers conturbo. Calco canonicus claudeo voveo quos.
        </p>
        <h3 id="tener-aliquam"
            node="[object Object]"
        >
          tener aliquam
        </h3>
        <p>
          Damno acsi curvo atavus depulso arguo cursim aetas color. Cursim tandem varietas suadeo vinco doloremque desipio delectus quia minus. Cognatus comes derideo.
        </p>
        <h3 id="cuppedia-statua"
            node="[object Object]"
        >
          cuppedia statua
        </h3>
        <p>
          Veritatis casso amitto denuncio deputo voro candidus earum debilito. Tantillus animi denique trepide thema pax cimentarius. Damnatio conicio absque cognomen confugo cur strenuus.
        </p>
        <h2 id="stabilis-ustulo"
            node="[object Object]"
        >
          stabilis ustulo
        </h2>
        <p>
          Ceno dolore defluo sortitus censura occaecati tui quibusdam cerno spero. Turbo atque agnosco comburo tenax vallum careo. Autem depulso mollitia benevolentia.
        </p>
        <h3 id="allatus-eligendi"
            node="[object Object]"
        >
          allatus eligendi
        </h3>
        <p>
          Artificiose quis verto supra vulgus acceptus demoror paens collum iusto. Articulus cicuta argumentum tendo sub amitto comis. Stultus verbera iste blanditiis creta conventus vespillo.
        </p>
        <h3 id="umbra-turba"
            node="[object Object]"
        >
          umbra turba
        </h3>
        <p>
          Contabesco crudelis adimpleo campana cogo crastinus cubicularis inflammatio. Iusto vulnero abeo constans reprehenderit tempus adsuesco optio. Stultus utique arcus.
        </p>
        <h3 id="clarus-cubitum"
            node="[object Object]"
        >
          clarus cubitum
        </h3>
        <p>
          Caput corona coniecto uterque dolores adamo sophismata ipsam amiculum. Aperte vado comminor degero. Aurum tyrannus claudeo ipsum toties ultra laboriosam creo.
        </p>
        <h2 id="spiritus-comes"
            node="[object Object]"
        >
          spiritus comes
        </h2>
        <p>
          Considero ustilo assumenda cura tricesimus adsum confido demulceo ipsam. Nesciunt nobis enim colligo abbas verto voveo. Ventosus conscendo beatus hic clarus deduco eius cimentarius.
        </p>
        <h3 id="veniam-clementia"
            node="[object Object]"
        >
          veniam clementia
        </h3>
        <p>
          Magnam causa sumo magni eaque timor consequuntur cruentus recusandae suus. Arto aspicio paulatim votum uredo consectetur vorago trepide ager. Cursus abscido bardus.
        </p>
        <h3 id="valens-defendo"
            node="[object Object]"
        >
          valens defendo
        </h3>
        <p>
          Infit aveho aliquid acsi clibanus totidem vestigium carbo modi adstringo. Aegre tardus demitto talis dolor administratio. Suggero cattus tumultus creber asperiores timor.
        </p>
        <h3 id="terminatio-tondeo"
            node="[object Object]"
        >
          terminatio tondeo
        </h3>
        <p>
          Stipes cohibeo vulgivagus sono cultellus. Auditor sono depopulo. Desipio subito culpa decerno attollo debitis adiuvo teneo pecco maxime.
        </p>
      </div>
    </div>
  </div>
</div>
`;
